library/src/cache.rs

```diff
@@ -121,7 +128,11 @@ impl UpdaterState {
         let reader = BufReader::new(file);
         // TODO: Now that we depend on serde_yaml for shorebird.yaml
         // we could use yaml here instead of json.
-        let state = serde_json::from_reader(reader)?;
+        let mut state: UpdaterState = serde_json::from_reader(reader)?;
+        if state.client_id.is_none() {
+            // Generate a client id if we don't already have one.
+            state.client_id = Some(generate_client_id());
```

Comment:
when does the next save happen? Are we guaranteed to persist the newly generated client_id? Is it possible for concurrent checks to generate separate client_ids?

Reply:
> when does the next save happen?
It depends on which function loads this. I added a `save` here to make this easier to reason about.

> Are we guaranteed to persist the newly generated client_id?
Added a `create_and_save_new` function to UpdaterState that is called in `load_or_new_on_error` when we construct a new UpdaterState.

> Is it possible for concurrent checks to generate separate client_ids?
No – the `with_config` wrapper acquires a lock before reading/writing state.